pkg/ext-proc/metrics/metrics_handler.go

```diff
@@ -0,0 +1,29 @@
+package metrics
+
+import (
+	"net"
+	"net/http"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+func StartMetricsHandler(port int) {
+	klog.Info("Starting metrics HTTP handler ...")
+
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.HandlerFor(
+		legacyregistry.DefaultGatherer,
+		promhttp.HandlerOpts{},
+	))
+
+	server := &http.Server{
```

Comment:
how are we securing this port?

Reply:
Moved to main.go with log.Fatal if it fails to listen.